test/units/executor/test_play_iterator.py

```diff
@@ -456,3 +458,53 @@ def test_play_iterator_add_tasks(self):
         # test a regular insertion
         s_copy = s.copy()
         res_state = itr._insert_tasks_into_state(s_copy, task_list=[MagicMock()])
+
+
+def test_play_iterator_skip_blank_name():
```

Comment:
For a straightforward test like this, I'd prefer an integration test. It's likely to be simpler and more readable. An integration test also won't hinder future refactoring since it shouldn't be dependent on internal implementation details.

Reply:
I'll look at changing it to an integration test. There are some other changes I think I need to make anyway.